experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/list.spec.ts

```diff
@@ -58,8 +58,8 @@ describe("List", () => {
 	/** Helper that creates a new SharedTree with the test schema and returns the root proxy. */
 	function createTree() {
 		// Consider 'initializeTreeWithContent' for readonly tests?
-		const view = createTreeView(schema, { numbers: [], strings: [] });
-		return view.root2(schema);
+		const view = createTreeView2(schema, { numbers: { "": [] }, strings: { "": [] } });
```

Comment:
What changed here is "any" got replaced with actual strong types. The old code does not type check (despite working at runtime) because SchemaAware's flexible mode never implemented the unwrapping rules for arrays. This was not done for a few reasons:
1. That code is getting replaced by the new schema based factory approach
2. The rules for when to do unwrapping/inlining of fields is planned to change as soon as we delete the old editable tree API since it has always been intended to use view schema info to make that choice, and thats impossible with the current state of things where the old editable tree does not have access to view schema. This OR is part of work toward that.